sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -659,12 +670,23 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
         AttributeReference("value", StringType)()), true)
     }
 
-    // Create the transform.
+    val plan = visitCommonSelectQueryClausePlan(
+      relation,
+      lateralView,
+      transformClause.namedExpressionSeq,
+      whereClause,
+      aggregationClause,
+      havingClause,
+      windowClause,
+      isDistinct = false)
+
     ScriptTransformation(
-      expressions,
+      // Create the transform, here we pass UnresolvedStart as ScriptTransform's input.
+      // In analyzer after child plan is resolved, we resolve UnresolvedStart as child's output.
```

Comment:
ditto: `// TODO: Remove this logic and see SPARK-34035`

Reply:
> ditto: `// TODO: Remove this param and see SPARK-34035`

Done